crates/containerd-shim-wasm/src/sandbox/containerd.rs

```diff
@@ -0,0 +1,176 @@
+            let req = with_namespace!(req, self.namespace);
+            let image = ContainersClient::new(self.inner.clone())
+                .get(req)
+                .await
+                .map_err(|err| ShimError::Containerd(err.to_string()))?
+                .into_inner()
+                .container
+                .ok_or_else(|| {
+                    ShimError::Containerd(format!(
+                        "failed to get image for container {}",
+                        container_name.to_string()
+                    ))
+                })?
+                .image;
+            Ok(image)
+        })
+    }
+
+    // load module will query the containerd store to find an image that has an OS of type 'wasm'
+    // If found it continues to parse the manifest and return the layers that contains the WASM modules
+    // and possibly other configuration layers.
+    pub fn load_modules(
+        &self,
+        containerd_id: impl ToString,
+    ) -> Result<(Vec<oci::WasmLayer>, Platform)> {
+        let image_name = self.get_image(containerd_id.to_string())?;
+        let digest = self.get_image_content_sha(image_name)?;
+        let manifest = self.read_content(digest)?;
+        let manifest = manifest.as_slice();
+        let manifest = ImageManifest::from_reader(manifest)?;
+
+        let image_config_descriptor = manifest.config();
+        let image_config = self.read_content(image_config_descriptor.digest())?;
+        let image_config = image_config.as_slice();
+
+        // the only part we care about here is the platform values
+        let platform: Platform = serde_json::from_slice(image_config)?;
+        match platform.architecture() {
+            Arch::Wasm => {
+                log::info!("found manifest with WASM OCI image format.");
+            }
+            _ => {
+                log::info!("manifest is not in WASM OCI image format");
+                return Ok(([].to_vec(), platform));
+            }
+        }
+
+        let layers = manifest
+            .layers()
+            .iter()
+            .filter(|x| !is_image_layer_type(x.media_type()))
+            .map(|config| {
+                self.read_content(config.digest()).map(|module| WasmLayer {
+                    config: config.clone(),
+                    layer: module,
+                })
+            })
+            .collect::<Result<Vec<_>>>()?;
+        Ok((layers, platform))
+    }
+}
+
+fn is_image_layer_type(media_type: &MediaType) -> bool {
+    match media_type {
+        MediaType::ImageLayer
+        | MediaType::ImageLayerGzip
+        | MediaType::ImageLayerNonDistributable
+        | MediaType::ImageLayerNonDistributableGzip
+        | MediaType::ImageLayerNonDistributableZstd
+        | MediaType::ImageLayerZstd => true,
+        MediaType::Other(s)
+            if s.as_str()
+                .starts_with("application/vnd.docker.image.rootfs.") =>
+        {
+            true
+        }
+        _ => false,
+    }
```

Comment:
The reason I didn't do just wasm layer types is because runtimes like spin/slight (probably others) will need to pass there configuration files.  By leaving this more open ended, initially, we allow for those other types. In the case of spin they also have a media type for a module that doesn't match the `application/vnd.bytecodealliance.wasm` format.  

I would like to tighten this up in the future once we get to the point where "everything is component" approach.  The extra config files, static files, etc would just be component that the runtime knows how to use.  But feels to early to be restrictive here.

I am not too concerned about this logic change all that often.  It is a borrowed, idea from containerd's [IsLayerType](https://github.com/containerd/containerd/blame/18c9e7ec4c692cc00b7ac75d86d60685c270d48a/images/mediatypes.go#L127) and that hasn't had a change in 4 years.

Reply:
We could add in the `Instance` trait a method to declare what layers it cares about (with a reasonable default impl).

Something like:
```rust
fn supported_layers_types() -> &'static [&'static str] {
    &["application/vnd.bytecodealliance.wasm.component.layer.v0+wasm"]
}
```

or maybe something like:
```rust
fn supports_layer_type(ty: MediaType) -> bool {
    match ty {
        MediaType::Other(s) => s.starts_with("application/vnd.bytecodealliance.wasm."),
        _ => false,
    }
}
```

If you think the current approach is better I'm happy with that as well, I don't want to block it over this :-)